src/features/events/components/EventChangeCampaignDialog.tsx

```diff
@@ -0,0 +1,206 @@
+import { Architecture, Close, Search } from '@mui/icons-material';
+import {
+  Alert,
+  Box,
+  Button,
+  CircularProgress,
+  Dialog,
+  DialogContent,
+  DialogTitle,
+  IconButton,
+  List,
+  ListItem,
+  TextField,
+  useMediaQuery,
+} from '@mui/material';
+import { makeStyles } from '@mui/styles';
+import { useContext, useState } from 'react';
+import { useRouter } from 'next/router';
+
+import theme from 'theme';
+import { Msg, useMessages } from 'core/i18n';
+import ZUISnackbarContext from 'zui/ZUISnackbarContext';
+import useCampaigns from 'features/campaigns/hooks/useCampaigns';
+import { ZetkinCampaign, ZetkinEvent } from 'utils/types/zetkin';
+import useEventMutations from '../hooks/useEventMutations';
+import messageIds from 'features/events/l10n/messageIds';
+
+interface EventActionButtonsProps {
+  event: ZetkinEvent;
+  isOpen: boolean;
+  close: () => void;
+}
+
+const useStyles = makeStyles(() => ({
+  list: {
+    listStyle: 'none',
+  },
+  listItem: {},
+}));
+
+const EventMoveDialog: React.FunctionComponent<EventActionButtonsProps> = ({
+  event,
+  isOpen,
+  close,
+}) => {
+  const classes = useStyles();
+  const messages = useMessages(messageIds);
+  const router = useRouter();
+  const { showSnackbar } = useContext(ZUISnackbarContext);
+  const [error, setError] = useState(false);
+
+  const { changeEventCampaign } = useEventMutations(
+    event.organization.id,
+    event.id
+  );
+
+  const fullScreen = useMediaQuery(theme.breakpoints.down('md'));
+
+  const [campaignFilter, setCampaignFilter] = useState('');
+  const [isLoadingCampaign, setIsLoadingCampaign] = useState(0);
+
+  const { data: campaigns } = useCampaigns(event.organization.id);
+  campaigns?.reverse();
+
+  const filteredCampaigns = campaigns
+    ?.filter((campaign) =>
+      campaign.title.toLowerCase().includes(campaignFilter)
+    )
+    .filter((campaign) => campaign.id !== event.campaign?.id);
+
+  const onSearchChange = (value: string) => {
+    setCampaignFilter(value);
+  };
+
+  const handleMove = async (campaign: ZetkinCampaign) => {
+    setIsLoadingCampaign(campaign.id);
+
+    try {
+      changeEventCampaign(campaign.id);
```

Comment:
If you modify `updateEvent()` and `changeEventCampaign()` to be async, you can `await` it here before showing the snackbar (or failing).